File: numeric/max/max.java
package numeric.max;

import java.util.List;
import java.util.ArrayList;

public class max
{
    public static int perform(List<Integer> values)
    {
        int current = 0;
        for (Integer value : values)
        {
            if (value > current)
            {
                current = value;
            }
        }
        return current;
    }

    public static void main(String[] args)
    {
        List<Integer> values = new ArrayList<Integer>();
        for (String arg : args)
        {
            values.add(Integer.parseInt(arg));
        }
        if (values.size() < 1)
        {
            values.add(15);
            values.add(10);
        }

        int max = perform(values);

        System.out.format("values: %s\nmax: %d\n", values.toString(), max);
    }
}

File: numeric/max/max.zig
const std = @import("std");

fn max(values: std.ArrayList(i32)) i32 {
    var current: i32 = 0;
    for (values.items) |value| {
        if (value > current) {
            current = value;
        }
    }
    return current;
}

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const allocator = gpa.allocator();
    defer _ = gpa.deinit();

    const args = try std.process.argsAlloc(allocator);
    defer std.process.argsFree(allocator, args);

    var values: std.ArrayList(i32) = .empty;
    defer values.deinit(allocator);

    if (args.len > 1) {
        for (args[1..]) |arg| {
            const t = std.fmt.parseInt(i32, arg, 10) catch 0;
            try values.append(allocator, t);
        }
    } else {
        try values.append(allocator, 15);
        try values.append(allocator, 10);
    }

    const maximum = max(values);

    std.debug.print("values: {any}\nmax: {}\n", 
        .{ values.items, maximum });
}
